### PR TITLE
Add the reconciler machinery for pod addressability

### PR DESCRIPTION
**Pull request by markusthoemmes**
# Changes

Ref knative/serving#10751

This is just creating the initial scaffolding with the expected resources so that subsequent diffs are more relevant to review.

/assign @julz @arturenault

### CODE REVIEW

File: pkg/reconciler/serverlessservice/controller.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	c := &reconciler{
+		kubeclient:            kubeclient.Get(ctx),
+		istioclient:           istioclient.Get(ctx),
+		virtualServiceLister:  virtualServiceInformer.Lister(),
+		destinationRuleLister: destinationRuleInformer.Lister(),
+	}
+	impl := sksreconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
+		logger.Info("Setting up ConfigMap receivers")
+		resyncIngressesOnConfigChange := configmap.TypeFilter(&config.Istio{})(func(string, interface{}) {
```
Comment (julz) [inline pkg/reconciler/serverlessservice/controller.go:56]: is this variable name correct here? I think it should be sthg like `resyncSKSOnConfigChange`

Comment (markusthoemmes) [reply pkg/reconciler/serverlessservice/controller.go:56]: Too much copy and paste... thanks!

**Review by julz — APPROVED** [review]
/lgtm

**Review by vagababov — APPROVED** [review]
/lgtm

**Pull request merged**